Dockerfile

```diff
@@ -25,26 +25,39 @@
 
 FROM centos:7 AS centos7-build-base
 
+# Requires DOCKER_BUILDKIT=1
+ARG BUILDARCH
+
 RUN set -x; yum -y update && \
 		yum -y install git systemd \
 		autoconf libtool libcurl-devel libtool-ltdl-devel openssl-devel yajl-devel \
 		gcc gcc-c++ make bison flex file systemd-devel zlib-devel gtest-devel rpm-build java-11-openjdk-devel \
 		expect rpm-sign zip wget && \
 		yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm && \
-		yum install -y cmake3 && \
-		ln -fs cmake3 /usr/bin/cmake && \
-		# Download and install CMake manually (zypper only gives us CMake 3.5, but we need >= 3.12)
-		wget https://github.com/Kitware/CMake/releases/download/v3.25.2/cmake-3.25.2-linux-x86_64.sh -O cmake.sh && \
-		(echo '4d98de8d605da676e71a889dd94f80c76abb377fade2f21e3510e62ece1e1ada  cmake.sh' | sha256sum --check) && \
-		bash cmake.sh --skip-license --prefix=/usr/local
+		case $BUILDARCH in \
+        'amd64') \
+                arch='x86_64' \
+                hash='4d98de8d605da676e71a889dd94f80c76abb377fade2f21e3510e62ece1e1ada' \
+                ;; \
+        'arm64') \
+                arch='aarch64' \
+                hash='73a35cab2174a3eb8f35083d55c80871185dc3808f3dae3558cd5fbdb29a4614' \
+                ;; \
+        *) \
+                echo 'unsupported' && false \
+                ;; \
+        esac && \
+	curl --location "https://github.com/Kitware/CMake/releases/download/v3.25.2/cmake-3.25.2-linux-$arch.sh" --output cmake.sh && \
+	(echo "$hash  cmake.sh" | sha256sum --check) && \
+	bash cmake.sh --skip-license --prefix=/usr/local
```

Comment:
There's actually a simpler way. Try adding the following at the top of the `Dockerfile`:
```
FROM alpine:latest AS cmake-amd64

ENV hash=4d98de8d605da676e71a889dd94f80c76abb377fade2f21e3510e62ece1e1ada
ADD https://github.com/Kitware/CMake/releases/download/v3.25.2/cmake-3.25.2-linux-x86_64.sh \
    /cmake.sh

FROM alpine:latest AS cmake-arm64

ENV hash=73a35cab2174a3eb8f35083d55c80871185dc3808f3dae3558cd5fbdb29a4614
ADD https://github.com/Kitware/CMake/releases/download/v3.25.2/cmake-3.25.2-linux-aarch64.sh \
    /cmake.sh

FROM cmake-${BUILDARCH} AS cmake-install
RUN set -xe; (echo "$hash  /cmake.sh" | sha256sum -c)
```
and then just use the following in each image:
```
COPY --from=cmake-install /cmake.sh /cmake.sh
RUN … && \
    bash /cmake.sh --skip-license --prefix=/usr/local
```

(aside: wish the [`--checksum` feature](https://docs.docker.com/engine/reference/builder/#verifying-a-remote-file-checksum-add---checksumchecksum-http-src-dest) were already available ­— then we'd be able to just do this:
```
FROM scratch AS cmake-amd64

ADD --checksum=sha256:4d98de8d605da676e71a889dd94f80c76abb377fade2f21e3510e62ece1e1ada \
    https://github.com/Kitware/CMake/releases/download/v3.25.2/cmake-3.25.2-linux-x86_64.sh \
    /cmake.sh

FROM scratch AS cmake-arm64

ADD --checksum=sha256:73a35cab2174a3eb8f35083d55c80871185dc3808f3dae3558cd5fbdb29a4614 \
    https://github.com/Kitware/CMake/releases/download/v3.25.2/cmake-3.25.2-linux-aarch64.sh \
    /cmake.sh

FROM cmake-${BUILDARCH} AS cmake-install
```
Oh, well…)

Reply:
Thanks -- added!